### DISCUSSION TITLE
closing AMQP connection client unexpectedly closed tcp connection

### DISCUSSION THREAD

**Discussion by libo1409** [rabbitmq/rabbitmq-server#6483] [Questions]
Use java program to connect to rabbitmq, exit after the application is started, see the mq log display close the connection, how to solve?

**Comment by lukebakken** [comment]
Your Java application must close the connection by calling `.close()` before exiting.

**Reply by libo1409** [reply]
Does that have anything to do with the maximum number of rabbitmq connections

**Reply by libo1409** [reply]
This log level is a warning in rabbitmq. If the application exits for a short time, mq does not close the connection properly, then this log will appear？

**Reply by lukebakken** [reply]
Have you confirmed that `.close()` is called when your application exits? Start there.

**Reply by lukebakken** [reply]
If you think there is an issue, provide code I can compile and run to see it. Anything else is just guessing which we do not have time to do.

**Reply by libo1409** [reply]
19/5000 
ok, let me check to see if I called .close(). thanks.

**Comment by michaelklishin** [comment]
See also [Connection Lifecycle Events](https://rabbitmq.com/logging.html#connection-lifecycle-events).
